p2p/module.go

```diff
@@ -334,13 +351,56 @@ func (m *p2pModule) handleAppData(data []byte) error {
 		return fmt.Errorf("decoding network message: %w", err)
 	}
 
+	if ok, err := m.checkNonce(networkMessage.GetNonce()); err != nil {
+		return fmt.Errorf("handling network data: %w", err)
+	} else if !ok {
+		// skip passing redundant message to application layer
+		return nil
+	}
+
+	// DISCUSS_THIS_COMMIT: is there an explicit reason for constructing a new
+	// `PocketEnvelop` literal with content rather than passing `networkMessage`?
+	// (e.g. avoid blindly passing additional fields as the protobuf type changes)
 	event := messaging.PocketEnvelope{
 		Content: networkMessage.Content,
 	}
 	m.GetBus().PublishEventToBus(&event)
 	return nil
 }
 
+// checkNonce ensures the nonce has not been observed with i the deuper's capacity
+// of recent messages. Adds the nonce to the deduper if it has not been observed.
+// DISCUSS(#278): Add more tests to verify this is sufficient for deduping purposes.
+func (m *p2pModule) checkNonce(nonce utils.Nonce) (ok bool, err error) {
```

Comment:
Can you rename `checkNonse` to something more "functional" like `wasNonceObserved` or `isDuplicateNonce` so there is no ambiguity as to what the boolean return value contains.